awscli/customizations/ecs/deploy.py

```diff
@@ -341,11 +342,26 @@ def describe_cd_resources(self):
             raise exceptions.ServiceClientError(
                 action='describe Code Deploy deployment group', error=e)
 
+        try:
+            dgp_info = self.deployment_group_details['deploymentGroupInfo']
+            deployment_config_name = dgp_info['deploymentConfigName']
+            self.deployment_config_details = self._client.get_deployment_config(
```

Comment:
@kdaily @allisaurus I'm concerned that it appears that we are adding an new API call `GetDeploymentConfig` to the default behavior of the command? If this is the case, it seems that it would cause issues where a user has a minimally scoped IAM policy to use this command (which would not include `GetDeploymentConfig` permissions) and if they were to upgrade their version of the CLI, they may start getting permissions issues. As an alternative, we could trigger this logic behind a flag (e.g. `--wait-for-traffic-reroute`) so that users have to opt into using the API call/wait logic. Thoughts?

Reply:
@kyleknap thanks for pointing this out, it's a valid concern. My one counter argument is that incorporating the required time for the different deployment steps was something this cmd was already doing previously, and customers could reasonably expect this to happen for free (vs. only w/ the use of a flag). Additionally, the [AWSCodeDeployDeployerAccess](https://docs.aws.amazon.com/codedeploy/latest/userguide/security_iam_id-based-policy-examples.html#managed-policies) managed policy includes `"codedeploy:Get*"`, so this population of users would not be effected by the addition of this call.

That said, I defer to your recommendation based on convention / previously seen issues elsewhere in `aws cli`. Are there other avenues (e.g., version release notes) where we could advertise the need for an additional read permission, to mitigate the minimally-scoped user concern in the absence of a flag?